src/kernel-install/90-loaderentry.install

```diff
@@ -106,41 +91,35 @@ install -g root -o root -m 0644 "$KERNEL_IMAGE" "$ENTRY_DIR_ABS/linux" || {
     exit 1
 }
 
-INITRD_OPTIONS=( "${@:${INITRD_OPTIONS_START}}" )
+shift "$INITRD_OPTIONS_SHIFT"
+for initrd; do
+    [ -f "$initrd" ] || continue
```

Comment:
BTW, I think this is a behaviour bug. If an *explicitly-supplied* initrd doesn't exist it's an error, and shouldn't get eaten.